### PR TITLE
3D Generation: SV3D Training, and Inference, for Multiview-consistent Video Generation

### PR DESCRIPTION
**Pull request by HaFred**
# What does this PR do?
This is an A+M implementation for [SV3D](https://huggingface.co/stabilityai/sv3d).
| Image 3  | 
| :-----------------: |
| Output Multiview Images (21x576x576) |

Please refer to the readme for more details.

### CODE REVIEW

File: examples/sv3d/configs/sv3d_u_train.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+environment:
+  mode: 0
+  seed: 42
+  distributed: False
+
+train:
+  epochs: 1000
+  temporal_only: True     # train only the temporal layers or the entire UNet
+  # pretrained:
+  pretrained: /mnt/disk4/fredhong/pretrained_ckpts/svd-d19a808f_embkeychaned_asinsv3d.ckpt  # as in the sv3d paper, finetuning from svd
```
Comment (Songyuanwei) [inline examples/sv3d/configs/sv3d_u_train.yaml:10]: 可以只写成pretrained_ckpts/svd-d19a808f_embkeychaned_asinsv3d.ckpt？最好不要出现具体详细的地址。

File: examples/sv3d/configs/sv3d_u_train.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      params:
+        p_mean: 1.0
+        p_std: 1.6
+
+    config_arch_toload_vanilla_sv3d_ckpt: False
+
+  dataset:
+    class_path: data.mulvideo_dataset.MulviewVideoDataset
+    init_args:
+      root_dir: /mnt/disk4/fredhong/mindone_master/examples/sv3d/itmh_training_data_9  # for overfitting exp
```
Comment (Songyuanwei) [inline examples/sv3d/configs/sv3d_u_train.yaml:32]: 同理

File: examples/sv3d/configs/sv3d_u_train.yaml (hunk 1 above)
Comment (HaFred) [reply examples/sv3d/configs/sv3d_u_train.yaml:10]: 好的

File: examples/sv3d/sgm/modules/train/trainer.py
Comment (Songyuanwei) [inline examples/sv3d/sgm/modules/train/trainer.py:1]: 这个和其他有些文件可以直接调用mindone下面的文件吗？

Comment (HaFred) [reply examples/sv3d/sgm/modules/train/trainer.py:1]: 可以的，我本地patch了一下wrapper所以加了这段代码。这个example下实际还是mindone下wrapper的功能，已更新。

File: examples/sv3d/sgm/modules/optimizers/adamw_zero1.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
+import numpy as np
```
Comment (CaitinZhao) [inline examples/sv3d/sgm/modules/optimizers/adamw_zero1.py:1]: why add this?

Comment (HaFred) [reply examples/sv3d/sgm/modules/optimizers/adamw_zero1.py:1]: Found that this a duplicate of `mindone/trainers/adamw_zero1.py`, removed.

**Review by CaitinZhao — APPROVED** [review]

File: examples/sv3d/train.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+python train.py \
+    --model_cfg configs/sampling/sv3d_u.yaml \
+    --train_cfg configs/sv3d_u_train.yaml \
+    --debug
```
Comment (CaitinZhao) [inline examples/sv3d/train.sh:4]: why debug？

Comment (HaFred) [reply examples/sv3d/train.sh:4]: This was used for debugging with no files dumped, should not be included here. Removed.

**Review by SamitHuang — APPROVED** [review]

**Review by vigo999 — APPROVED** [review]

**Pull request merged**